File: tests/benchmarks/resolution/fixtures/objc/Validators.m
#import "Validators.h"

@implementation Validators

+ (BOOL)isValidEmail:(NSString *)email {
    return [email containsString:@"@"] && [email containsString:@"."];
}

+ (BOOL)isValidName:(NSString *)name {
    return name.length >= 2;
}

+ (BOOL)validateUser:(NSString *)name email:(NSString *)email {
    return [self isValidName:name] && [self isValidEmail:email];
}

@end

File: tests/fixtures/bare-call-scope/Validators.cs
namespace App
{
    public static class Validators
    {
        public static bool ValidateUser(string email)
        {
            return IsValidEmail(email);
        }

        public static bool IsValidEmail(string email)
        {
            return email.Contains("@");
        }
    }

    public static class Formatters
    {
        // Same method name as Validators.IsValidEmail — must NOT cross-resolve.
        public static bool IsValidEmail(string value)
        {
            return false;
        }

        public static string FormatName(string name)
        {
            return name.Trim();
        }
    }
}
